lib/src/providers/accountProvider.dart

```diff
@@ -0,0 +1,162 @@
+            log(response.data.toString());
+            throw ApiException(response.data['error']);
+          }
+
+          Map<String, dynamic> payload = response.data["payload"];
+
+          // save access token (refresh token is auto-saved by CookieManager above)
+          if (payload["accesstoken"] != null) {
+            await storage.write(
+              key: 'jwt_access',
+              value: payload[0]['accessToken'],
+            );
+            jwt = payload[0]['accessToken'];
+            notifyListeners();
+          }
+          handler.resolve(response);
+        },
+      ),
+    );
+    await this.getFeed();
+    await this.getSections();
+
+    working = false;
+    notifyListeners();
+  }
+
+  CookieJar get cookieJar => this.api.cookieJar;
+
+  Future<List<FeedItem>?> getFeed() async {
+    Uri uri = Uri.http(Config.baseEndpoint!, "/api/v1/feed/list");
+
+    final response = await this.api.getUri(
+          uri,
+          options: dioOptions.Options(
+            headers: {"Authorization": "Bearer ${this.jwt}"},
+          ),
+        );
+
+    if (response.statusCode != 200) {
+      throw HttpException(response.statusCode.toString());
+    }
+
+    Map<String, dynamic> data = response.data;
+
+    if (data['error'] != null) {
+      log(data.toString());
+      throw ApiException(data['error']);
+    }
+
+    if (List.from(data["payload"]["feed"]).isEmpty) {
+      feedItems = [];
+      notifyListeners();
+      return feedItems;
+    }
+
+    feedItems =
+        data['payload'].map<FeedItem>((e) => FeedItem.fromJSON(e)).toList();
+    feedItems!.sort((a, b) => a.postDate.compareTo(b.postDate));
+    notifyListeners();
+    return feedItems;
+  }
+
+  Future<List<SectionItem>?> getSections() async {
+    Uri uri = Uri.http(Config.baseEndpoint!, "/api/v1/library/search");
+
+    final response = await this.api.getUri(
+          uri,
+          options: dioOptions.Options(
+            headers: {"Authorization": "Bearer ${this.jwt}"},
+          ),
+        );
+
+    if (response.statusCode != 200) {
+      throw HttpException(response.statusCode.toString());
+    }
+
+    Map<String, dynamic> data = response.data;
+
+    if (data['error'] != null) {
+      log(data.toString());
+      throw ApiException(data['error']);
+    }
+
+    if (data["payload"]["library"].isEmpty) {
+      sections = [];
+      notifyListeners();
+      return sections;
+    }
+
+    sections = data['payload'].map((e) => SectionItem.fromJSON(e)).toList();
+    notifyListeners();
+    return sections;
+  }
+
+  Future<void> addSectionItem(SectionItem sectionItem) async {
+    Uri uri = Uri.http(Config.baseEndpoint!, "/api/v1/library/add");
+
+    final response = await this.api.postUri(uri,
+        options: dioOptions.Options(
+          headers: {"Authorization": "Bearer ${this.jwt}"},
+        ),
+        data: {"course": sectionItem.fullCode, "section": sectionItem.crn});
+
+    if (response.statusCode != 200) {
```

Comment:
Is there duplicated logic in these functions that could be abstracted?

Reply:
Can probably abstract the options within the class itself.